src/commcare_cloud/commands/ansible/helpers.py

```diff
@@ -35,27 +35,38 @@ def _build_env(self, args):
         return env
 
 
-def get_common_ssh_args(environment, use_pem=False):
+def get_common_ssh_args(cmd_parts, environment, use_factory_auth=False):
     strict_host_key_checking = environment.public_vars.get('commcare_cloud_strict_host_key_checking', True)
 
     common_ssh_args = []
-    if use_pem:
-        pem = environment.public_vars.get('commcare_cloud_pem', None)
-        if pem:
-            common_ssh_args.extend(['-i', pem])
+
+    if use_factory_auth:
+        cmd_parts, common_ssh_args = add_factory_auth_cmd(environment, common_ssh_args, cmd_parts)
```

Comment:
Won't this clobber the `cmd_parts` that was passed in? My reading is that the passed in variable contains lots of crucial stuff about the command to be run that can't be thrown away. I think this kind of error (if I'm following what's going on here correctly) is a good demonstration of why I think it's simpler to reason about not to pass in `cmd_parts` and keep the function signature the way it was previously.

Reply:
Doesn't error, but updated this to make more readable here:
https://github.com/dimagi/commcare-cloud/pull/1702/commits/f70a5f2d756f90ed5d88422d9e4af563fdb2fe46